.tools/validate.go

```diff
@@ -39,6 +43,35 @@ var DefaultRules = []ValidateRule{
 		return vr
 	},
 	// TODO add something for the cleanliness of the c.Subject
+	func(c CommitEntry) (vr ValidateResult) {
+		vr.CommitEntry = c
+		buf := bytes.NewBuffer([]byte{})
+		args := []string{"git", "show", "--check", c["commit"]}
+		vr.Msg = strings.Join(args, " ")
+		cmd := exec.Command(args[0], args[1:]...)
+		cmd.Stdout = buf
+		cmd.Stderr = os.Stderr
+		if err := cmd.Run(); err != nil {
+			vr.Pass = false
+			vr.Detail = string(buf.Bytes())
+			return vr
+		}
+		vr.Pass = true
+		return vr
+	},
+	func(c CommitEntry) (vr ValidateResult) {
+		return ExecTree(c, "go", "vet", "./...")
+	},
+	func(c CommitEntry) (vr ValidateResult) {
+		return ExecTree(c, "go", "fmt", "./...")
+	},
+	func(c CommitEntry) (vr ValidateResult) {
+		vr = ExecTree(c, os.ExpandEnv("$HOME/gopath/bin/golint"), "./...")
```

Comment:
On Mon, Oct 05, 2015 at 12:20:58PM -0700, Vincent Batts wrote:
> > +		vr = ExecTree(c, os.ExpandEnv("$HOME/gopath/bin/golint"), "./...")
> 
> but this is what we're doing in travis, and seems cleaner there.

You're not doing it for every commit in Travis, so you'd miss cases
where one commit breaks the format and a subsequent commit fixed it.